Sources/SwiftSDKGenerator/Queries/CMakeBuildQuery.swift

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2022-2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import GeneratorEngine
+import struct SystemPackage.FilePath
+
+@Query
+struct CMakeBuildQuery {
+  let sourcesDirectory: FilePath
+  let outputBinarySubpath: [FilePath.Component]
+  let options: String
+
+  func run(engine: Engine) async throws -> FilePath {
+    try await Shell.run(
+      """
+      cmake -B build -G Ninja -S llvm -DCMAKE_BUILD_TYPE=Release \(self.options)
+      """,
+      currentDirectory: self.sourcesDirectory
+    )
+
+    let buildDirectory = self.sourcesDirectory.appending("build")
+    try await Shell.run("ninja", currentDirectory: buildDirectory)
+
+    return self.outputBinarySubpath.reduce(into: buildDirectory) { $0.append($1) }
```

Comment:
```suggestion
    return buildDirectory.appending(outputBinarySubpath.compontents)
```

Reply:
This was mostly about making it explicit that the query takes relative paths as arguments and not absolute ones.

Reply:
Yes, I was reading the docs for FilePath just now and I wondered if that was what you were intending to do, to make sure components such as `bin/lld` aren't mixed up with full paths.

Reply:
I've added a doc comment on that `CMakeBuildQuery` property explicitly mentioning relative paths, which I hope will clarify things a bit.